workspaces/ui-v2/src/pages/docs/DocumentationRootPage/DocumentationRootPage.tsx

```diff
@@ -42,10 +44,21 @@ export function DocumentationRootPage() {
     endpointsState.data || []
   );
 
-  const grouped = useMemo(() => groupBy(filteredEndpoints, 'group'), [
-    filteredEndpoints,
-  ]);
-  const tocKeys = Object.keys(grouped).sort();
+  const groupedEndpoints = useMemo(() => {
+    const commonStart = findLongestCommonPath(
+      filteredEndpoints.map((endpoint) => endpoint.fullPath)
+    );
+    const endpointsWithGroups = filteredEndpoints.map((endpoint) => ({
+      ...endpoint,
+      // If there is only one endpoint, split['/'][1] returns undefined since
+      // commonStart.length === endpoint.fullPath.length
+      group: endpoint.fullPath.slice(commonStart.length).split('/')[1] || '',
```

Comment:
spectacle exposes the pathComponents as an array as well as a string. The array is for use cases like this. What is a human-readable way to describe the desired sorting and grouping of this ui?

Reply:
`group` is a string which is where the endpoint should be grouped in the documentation page.

E.g.
`/api/list`
`/api/todos`
`/api/todos/:todoId`

The common start is `/api` and each would be grouped under the next path parameter (in this case, there would be two groupings - todos and list)

Reply:
Cool we can revisit later but this is definitely less broken than before 😓 please use the spectacle-provided path components if possible

Reply:
Oh I just saw this - I'll follow this PR up with a separate fix (changing the function signature)